operator/pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DMNode is a specification for a DMNode resource
+type DMNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              DMNodeSpec   `json:"spec"`
+	Status            DMNodeStatus `json:"status"`
+}
+
+// DMNodeSpec is the spec for a DMNode resource
+type DMNodeSpec struct {
+	ContainerSpec
+	Service          DMNodeService                        `json:"service,omitempty"`
+	Timezone         string                               `json:"timezone,omitempty"`
+	Replicas         int32                                `json:"replicas,omitempty"`
+	Affinity         *corev1.Affinity                     `json:"affinity,omitempty"`
+	NodeSelector     map[string]string                    `json:"nodeSelector,omitempty"`
+	StorageClassName string                               `json:"storageClassName,omitempty"`
+	Tolerations      []corev1.Toleration                  `json:"tolerations,omitempty"`
+	Annotations      map[string]string                    `json:"annotations,omitempty"`
+	PVReclaimPolicy  corev1.PersistentVolumeReclaimPolicy `json:"pvReclaimPolicy,omitempty"`
+	Source           DMSource                             `json:"source,omitempty"`
+	NodeConfig       DMWorkerConfig                       `json:"nodeConfig,omitempty"`
```

Comment:
The DMWorkerConfig, DMMasterConfig, DMTask, etc. types definition should reuse the ones in dm code, however, the deepcopy functions cannot be generated if just import them in this file, may need to check further if deepcopy issue can be fixed so that no need to re-define these types here again to reduce maintain effort.